### PR TITLE
1009 - The Conjur HTTP connector makes use of the generic connector 

### PR DESCRIPTION
**Pull request by BradleyBoutcher**
Several changes were made to the Conjur Connector to allow use of the generic connector.

#### What does this PR do (include background context, if relevant)?
Since @jonahx created the generic connector, allowing the injection of 
headers to create custom connectors for new services, we are 
attempting to reduce our code base and streamline our existing 
connectors. To that end, several changes were made to the
Conjur connector.

- The connector.go file was removed from the conjur directory,
with the NewConnectorConstructor taking the place of its functionality.

- The GetHTTPPlugin method in the Conjur Connector now defines the headers
used for authorization, such as the token.

- Relocaed the forceSSL property, per generic connector standards.

#### What ticket does this PR close?
#1009 

#### Where should the reviewer start?
secretless-broker/internal/plugin/connectors/http/conjur/plugin.go

#### What is the status of the manual tests?
Have you run the following manual tests to verify existing functionality continues to function as expected?
- [X] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/manual/keychain_provider)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

If this feature does not have any/sufficent automated tests, have you created/updated a folder in `test/manual` that includes:
- [x] An updated README with instructions on how to manually test this feature
- [x] Utility `start` and `stop` scripts to spin up and tear down the test environments
- [x] A `test` script to run some basic manual tests (optional; if does not exist, the README should have detailed instructions)

### CODE REVIEW

File: internal/plugin/connectors/http/conjur/plugin.go [hunk 1]
```diff
@@ -16,16 +18,26 @@ func PluginInfo() map[string]string {
...
-}
-
 // GetHTTPPlugin is required as part of the Secretless plugin spec for HTTP
 // connector plugins. It returns the HTTP plugin.
 func GetHTTPPlugin() http.Plugin {
-	return http.NewConnectorFunc(NewConnector)
+	newConnector, err := generichttp.NewConnectorConstructor(
+		&generichttp.ConfigYAML{
+			CredentialValidations: map[string]string{
+				// Unsure
```
Comment (izgeri) [inline internal/plugin/connectors/http/conjur/plugin.go]: are you looking for the standard format of the Conjur token? that's probably worth asking about in Slack and adding here before this gets merged

Comment (BradleyBoutcher) [reply internal/plugin/connectors/http/conjur/plugin.go]: Good catch - I checked with Jonah yesterday and forgot to remove it.

Comment (izgeri) [reply internal/plugin/connectors/http/conjur/plugin.go]: if it helps it looks like it's just a JWT: https://docs.conjur.org/Latest/en/Content/Get%20Started/cryptography.html?Highlight=cryptography#Authenticationtokens

Comment (izgeri) [reply internal/plugin/connectors/http/conjur/plugin.go]: whoops, missed your comment. 👍

File: test/connector/http/conjur/secretless.yml [hunk 2]
```diff
@@ -11,3 +11,4 @@ services:
     config:
       authenticateURLsMatching:
         - ^http\:\/\/conjur\/
+      forceSSl: false
```
Comment (doodlesbykumbi) [inline test/connector/http/conjur/secretless.yml]: Missing newline at EOF

File: internal/plugin/connectors/http/conjur/plugin.go [hunk 3]
```diff
@@ -16,16 +18,23 @@ func PluginInfo() map[string]string {
...
-}
-
 // GetHTTPPlugin is required as part of the Secretless plugin spec for HTTP
 // connector plugins. It returns the HTTP plugin.
 func GetHTTPPlugin() http.Plugin {
-	return http.NewConnectorFunc(NewConnector)
+	newConnector, err := generichttp.NewConnectorConstructor(
+		&generichttp.ConfigYAML{
+			Headers: map[string]string{
+				"Authorization": "Token token={{ `%s` .token | base64 }}",
```
Comment (doodlesbykumbi) [inline internal/plugin/connectors/http/conjur/plugin.go]: Since the token value needs to be surrounded by quotation marks I'd suggest the following:
```
"Authorization": `Token token="{{ .token | base64 }}"`,
```

File: test/connector/http/conjur/secretless.yml [hunk 4]
```diff
@@ -11,3 +11,5 @@ services:
...
```
Comment (jonahx) [inline test/connector/http/conjur/secretless.yml]: typo: `forceSSL`

File: internal/plugin/connectors/http/conjur/plugin.go [hunk 5]
```diff
@@ -1,7 +1,9 @@
 package conjur
 
 import (
-	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"log"
+
+	generichttp "github.com/cyberark/secretless-broker/internal/plugin/connectors/http/generic"
```
Comment (jonahx) [inline internal/plugin/connectors/http/conjur/plugin.go]: I had originally named the package `generichttp` but after feedback from Srdjan and (I thought) you  we decided to go with the name `generic` since we thought the "http" would be clear from context.

So I think either:
1. We revisit that decision and change the package.
2. Keep the package name and don't do the rename here.

At this point I'd lean toward 2.

Generally speaking, you should only be renaming a package because it has a conflict with another package.  Otherwise it's telling you that the package name is not good enough, and you should fix the root problem (assuming it's code we own).

File: test/connector/http/conjur/secretless.yml [hunk 6]
```diff
@@ -11,3 +11,5 @@ services:
     config:
       authenticateURLsMatching:
         - ^http\:\/\/conjur\/
+      forceSSL: false
+
```
Comment (sgnn7) [inline test/connector/http/conjur/secretless.yml]: Drive-by comment: both ymls have a stray newline at the end of the files
